### PR TITLE
[kmac, doc] Update `MSG_FIFO` description

### PR DESCRIPTION
**Pull request by ballifatih**
Spawned from the thread https://github.com/lowRISC/opentitan/pull/19554#discussion_r1314309770.

The description of `MSG_FIFO` states that when writing bytes/words to it, SW can ignore the word addresses but should correctly provide the least significant 2 bits of the address, so that MSG_FIFO can recognize the position of a byte within word. I think this is incorrect (a test by @jadephilipoom also points to this direction).

What really happens is that, even if SW ignores the least significant 2 bits of the address and directly writes a byte, its alignment is internally corrected by prim_packer primitive inside MSG_FIFO. As prim_packer is receiving input bytes (in order to accumulate them to word, and then release), it only takes the masking bits that come from TL-UL transaction into account. prim_packer keeps track of previously written bytes/words and uses this information to correctly align the position of the incoming byte within words.

In short, SW can always simply write to the beginning address of MSG_FIFO, and ignore the address calculation details. Even if SW switched arbitrarily between bytes/half-words/words while writing to a fixed address in MSG_FIFO range, it would work.

Therefore this PR updates the doc.

### CODE REVIEW

**Review by jadephilipoom — APPROVED** [review]

File: hw/ip/kmac/data/kmac.hjson [hunk 1]
```diff
@@ -1010,9 +1010,11 @@
         byte-write: "true"
         desc: '''Message FIFO.
 
-              Any write to this window will be appended to the FIFO. Only lower
-              2 bits `[1:0]` of the address matter to writes within the window
-              in order to handle with sub-word writes.
+              Any write operation to this window will be appended to MSG_FIFO. SW can
```
Comment (vogelpi) [inline hw/ip/kmac/data/kmac.hjson:1013]: I always recommend to switch to 1 sentence per line whenever we change hjson and markdown files as this will simplify future diffs. The 100 or whatever chars per line is something we primarily did early in the project.

**Review by vogelpi — APPROVED** [review]
Thanks @ballifatih for the PR and the clear description / explanation, LGTM!

**Comment by vogelpi** [comment]
CHANGE AUTHORIZED: hw/ip/kmac/data/kmac.hjson

This PR just changes the documentation but not the RTL or register map or similar.

**Comment by vogelpi** [comment]
@jadephilipoom : would you mind adding an AUTHORIZE comment similar to mine please? It's required for merging this PR.

**Comment by jadephilipoom** [comment]
CHANGE AUTHORIZED: hw/ip/kmac/data/kmac.hjson

As Pirmin said, this PR only changes the documentation to match the RTL.

File: hw/ip/kmac/data/kmac.hjson (hunk 1 above)
Comment (ballifatih) [reply hw/ip/kmac/data/kmac.hjson:1013]: I was abiding by that rule for doc changes, but I was not aware it also applied to hjson files. I will keep that in mind for future commits.
